src/ir_def/check_ir.ml

```diff
@@ -1057,7 +1057,10 @@ and check_dec env dec  =
   | LetD (pat, exp) ->
     ignore (check_pat_exhaustive env pat);
     check_exp env exp;
-    typ exp <: pat.note
+    typ exp <: pat.note;
+    check env pat.at
+      Ir_utils.(expected_irrefutable dec ==> is_irrefutable pat)
+      "refutable pattern in import"
```

Comment:
I  don't think you should be checking this here but in typing.ml, where all other pats are checked.
```suggestion
```
And then remove the expected_irrefutable hack below

Reply:
I think I a bit dense today. `typing.ml` looks at AST, and the `LetD` is being added to IR.
But we can see the pattern and the `ImportE` in the AST, which should be good enough. Is this what you mean?